frontend_app_logbook/src/component/Logbook.jsx

```diff
@@ -0,0 +1,316 @@
+    }
+  }
+
+  componentDidMount () {
+    const { props } = this
+    if (props.isNewContentRevision) this.loadLogbookContent()
+
+    if (props.content.current_revision_type === 'creation') {
+      const newLogbook = { entries: [] }
+      this.setState({
+        logbookState: LOGBOOK_STATE.LOADED,
+        logbookInitiallyLoaded: true,
+        saveRequired: true,
+        logbook: newLogbook
+      })
+    }
+  }
+
+  async componentDidUpdate (prevProps) {
+    const { state, props } = this
+    if (
+      (props.content.current_revision_id !== prevProps.content.current_revision_id) ||
+      (props.isNewContentRevision && prevProps.isNewContentRevision !== props.isNewContentRevision)
+    ) {
+      this.loadLogbookContent()
+    }
+
+    if (state.saveRequired) {
+      this.save(state.logbook)
+      this.setState({ saveRequired: false })
+    }
+  }
+
+  async loadLogbookContent () {
+    this.setState({ logbookState: LOGBOOK_STATE.LOADING })
+    const { props } = this
+
+    try {
+      const fetchRawFileContent = await handleFetchResult(
+        await getRawFileContent(
+          props.config.apiUrl,
+          props.content.workspace_id,
+          props.content.content_id,
+          props.content.current_revision_id,
+          props.content.label + LOGBOOK_FILE_EXTENSION
+        ),
+        true
+      )
+
+      if (fetchRawFileContent.apiResponse.ok && fetchRawFileContent.body.entries) {
+        this.setState({
+          logbookState: LOGBOOK_STATE.LOADED,
+          logbookInitiallyLoaded: true,
+          logbook: fetchRawFileContent.body
+        })
+      } else {
+        console.error(fetchRawFileContent)
+        this.setState({ logbookState: LOGBOOK_STATE.ERROR })
+      }
+    } catch (e) {
+      console.error(e)
+      this.setState({ logbookState: LOGBOOK_STATE.ERROR })
+    }
+  }
+
+  handleShowPopIn = (entry) => {
+    this.setState({
+      entryToEdit: entry,
+      showEditPopIn: true
+    })
+  }
+
+  handleHidePopIn = () => {
+    this.setState({
+      entryToEdit: {},
+      showEditPopIn: false
+    })
+  }
+
+  handleAddOrEditEntry = (entry) => {
+    this.setState(prevState => {
+      const newLogbook = entry.id
+        ? replaceEntryInLogbook(prevState.logbook, entry)
+        : addEntryToLogbook(prevState.logbook, entry)
+      return {
+        showEditPopIn: false,
+        logbookState: LOGBOOK_STATE.SAVING,
+        saveRequired: true,
+        logbook: newLogbook
+      }
+    })
+  }
+
+  handleRemoveEntry = (entry) => {
+    this.setState(prevState => {
+      const newLogbook = removeEntryFromLogbook(prevState.logbook, entry)
+      return {
+        logbookState: LOGBOOK_STATE.SAVING,
+        saveRequired: true,
+        logbook: newLogbook
+      }
+    })
+  }
+
+  async save (newLogbook) {
+    const { props } = this
+    const sortedLogbook = newLogbook
+    sortedLogbook.entries = newLogbook.entries.toSorted((a, b) => new Date(b.datetime) - new Date(a.datetime))
```

Comment:
by creating sortedLogbook that way you create a reference to newLogbook

So the affectation will update both newLogbook and sortedLogbook

prefer:
```jsx
const sortedLogbook = { ...newLogbook }
```

Reply:
or, better,
```js
const sortedLogbook = { entries: newLogbook.entries.toSorted((a, b) => new Date(b.datetime) - new Date(a.datetime)) }
```

Reply:
fixed as suggested in the second suggestion